src/models/components/prior_encoder.py

```diff
@@ -0,0 +1,51 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.distributions import Distribution
+from torch.distributions.normal import Normal
+
+from ..abc.prior import Prior
+from ..abc._types import _t_or_any, _tensor_or_any
+
+
+class Prior(Prior):
```

Comment:
上記の提案を採用しないのであればクラス名を変更してください。
```suggestion
class PriorEncoder(Prior):
```